Закончил выполнение задания 7-го спринта.

MaximusDenisov-YP opened the pull request:
- Добавил реализацию TaskManager - FileBackedTaskManager, который имеет возможность считывать, записывать, хранить состояния задач в файле src/main/resources/autosave.csv
- Обновил .gitignore, чтобы не уходили лишние директории, так же удалил из кэша каталог .idea
- Добавил кастомное исключение ManagerSaveException.java
- Удалил лишний файл Main.java, в котором содержался пользовательский сценарий предыдущего задания
- Пользовательский сценарий данного задания находится в классе FileBackedTaskManager.java, как и указано в доп. задании

Commit:
Закончил выполнение задания 7-го спринта.

- Добавил реализацию TaskManager - FileBackedTaskManager, который имеет возможность считывать, записывать, хранить состояния задач в файле src/main/resources/autosave.csv
- Обновил .gitignore, чтобы не уходили лишние директории, так же удалил из кэша каталог .idea
- Добавил кастомное исключение ManagerSaveException.java
- Удалил лишний файл Main.java, в котором содержался пользовательский сценарий предыдущего задания
- Пользовательский сценарий данного задания находится в классе FileBackedTaskManager.java, как и указано в доп. задании

avfyodorov commented on src/main/ru/kanban/manager/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,277 @@
+import java.nio.file.StandardOpenOption;
+import java.util.ArrayList;
+
+public class FileBackedTaskManager extends InMemoryTaskManager implements TaskManager {
```
implements TaskManager

Повторно реализовывать интерфейс не нужно. Он уже реализован в классе-предке. 
Здесь достаточно будет переписать только те методы, у которых изменилось поведение.

avfyodorov commented on src/main/ru/kanban/manager/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,277 @@
+                }
+            }
+        } catch (IOException e) {
+            throw new RuntimeException(e);
```
Тут немного   не так. Перехватываете IOException и сами пробрасываете
```java
  ...........  
      } catch (IOException e) {
            throw new ManagerSaveException("Ошибка при чтении файла: " + e.getMessage());
        }
```

avfyodorov commented on src/main/ru/kanban/manager/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,277 @@
+    }
+
+    @Override
+    public ArrayList<Task> getTasks() {
```
Однострочные методы, состоящие из одной super, не добавляют никакой новой функциональности. Поэтому они не нужны. От них можно спокойно избавиться.

avfyodorov reviewed (commented):
Добрый день, Максим!

Практически всё сделано, осталось чуть  поправить.

Commit:
Исправил замечания по практической работе 7-го спринта.

- Убрал лишний implements интерфейса, так как он уже есть в наследуемом классе.
- Убрал однострочные методы super, так как механизм наследования уже реализует эту логику.
- Исправил отбрасывание кастомного исключения в методе loadFromFile.

avfyodorov reviewed (commented):
Добрый день, Максим!

Замечаний нет.
Работа принята.
